update Cabal installation instruction

neosimsim opened the pull request:
Since Cabal 3.0 `v2-*` commands have become the default and packages are
no longer globally installed by default, see

https://www.haskell.org/cabal/users-guide/nix-local-build-overview.html

This requires required libraries (e.g. ieee754) to be registered
manually in the GHC environment.

neosimsim commented:
I'm not sure if the old instructions should be kept. I use Cabal 3.0 exclusively for a while now, but I'm not sure if that's the case for everyone.

nad commented:
We still support GHC 8.0.2, and I don't think we should force users to upgrade cabal-install.

neosimsim commented:
Okay fine, makes sense. Would it be useful to split the instruction into to parts, then? One for Cabal <2.4 and one for Cabal >=2.4? I would be glad to do so.

The thing is, that it took me a while to get Agda working with Cabal 3.0.0.0. After trying stack and nix
I finally found a way do to it with Cabal. Maybe others find it helpful, too.

neosimsim committed:
Add installation instruction for libraries using Cabal new-style builds

Since Cabal 3.0 v2-* commands have become the default and packages are
no longer globally installed by default, see

https://www.haskell.org/cabal/users-guide/nix-local-build-overview.html

This requires required libraries (e.g. ieee754) to be registered
manually in the GHC environment.

https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/packages.html?highlight=environment#package-environments

jrp2014 commented:
This one bit me too.  Out of the box, `make` compiles OK, provided that you fiddle around installing the right `icu` library (which on a Mac with `brew` requires some path fiddling) but `make test` fails in an ugly way, missing the `ieee754` library.  It would be good to fix this in the `Makefile` if that didn't break `stack`.

Incidentally, the interaction (and perhaps other) test fail, when using cabal, because the output contains extra lines like:
```
Loaded package environment from /Users/user/.ghc/x86_64-darwin-8.8.2/environments/default
```

asr commented on doc/user-manual/getting-started/installation.rst:
```diff
@@ -56,6 +56,50 @@ This can, in some cases, give a noticeable speedup.
+
+.. NOTE::
+
+  Actually it is not necessary to register the Agda libraries,
```
libraries -> library

asr commented:
> This one bit me too. Out of the box, `make` compiles OK, provided that you fiddle around installing the right `icu` library (which on a Mac with `brew` requires some path fiddling) but `make test` fails in an ugly way, missing the `ieee754` library. It would be good to fix this in the `Makefile` if that didn't break `stack`.
> 
> Incidentally, the interaction (and perhaps other) test fail, when using cabal, because the output contains extra lines like:
> 
> ```
> Loaded package environment from /Users/user/.ghc/x86_64-darwin-8.8.2/environments/default
> ```

@jrp2014, please uses #4216 for reporting problems in the `Makefile` with recent versions of `cabal-install`.

asr commented:
Thanks!

neosimsim commented:
@asr Thanks for fixing the typo and sorry for not responding until now.
